src/tribol/mesh/MfemData.hpp

```diff
@@ -810,6 +810,200 @@ class MfemMeshData
+
+  /**
+   * @brief Sets the penalty scale for the first registered Tribol mesh
+   * 
+   * @param scale Penalty scale value for the first registered Tribol mesh
+   */
+  void SetMesh1PenaltyScale(real scale) { scale_1_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Sets the penalty scale for the second registered Tribol mesh
+   * 
+   * @param scale Penalty scale value for the second registered Tribol mesh
+   */
+  void SetMesh2PenaltyScale(real scale) { scale_2_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Get the penalty scale for the first registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh1PenaltyScale() const { return scale_1_.get(); }
+
+  /**
+   * @brief Get the penalty scale for the second registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh2PenaltyScale() const { return scale_2_.get(); }
+
+  /**
+   * @brief Sets the rate penalty for the first registered Tribol mesh
+   * 
+   * @param penalty Rate penalty value for the first registered Tribol mesh
+   */
+  void SetMesh1RatePenalty(real penalty) { rate_penalty_1_ = std::make_unique<double>(penalty); }
+
+  /**
+   * @brief Sets the rate penalty for the second registered Tribol mesh
+   * 
+   * @param penalty Rate penalty value for the second registered Tribol mesh
+   */
+  void SetMesh2RatePenalty(real penalty) { rate_penalty_2_ = std::make_unique<double>(penalty); }
+
+  /**
+   * @brief Get the rate penalty for the first registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh1RatePenalty() const { return rate_penalty_1_.get(); }
+
+  /**
+   * @brief Get the rate penalty for the second registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh2RatePenalty() const { return rate_penalty_2_.get(); }
+
+  /**
+   * @brief Sets the rate penalty scale for the first registered Tribol mesh
+   * 
+   * @param scale Rate penalty scale value for the first registered Tribol mesh
+   */
+  void SetMesh1RateScale(real scale) { rate_scale_1_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Sets the rate penalty scale for the second registered Tribol mesh
+   * 
+   * @param scale Rate penalty scale value for the second registered Tribol mesh
+   */
+  void SetMesh2RateScale(real scale) { rate_scale_2_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Get the rate penalty scale for the first registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh1RateScale() const { return rate_scale_1_.get(); }
+
+  /**
+   * @brief Get the rate penalty scale for the second registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh2RateScale() const { return rate_scale_2_.get(); }
+
+  /**
+   * @brief Get a pointer to the element thickness array for the first Tribol
+   * registered mesh
+   *
+   * @return const real* 
+   */
+  const real* GetRedecompElemThickness1() const
```

Comment:
Double check that this routine/check works with zero element redecomp meshes. There were API calls that check if this is a non-null pointer, but looks like it will be for zero element redecomp meshes which will kill a simulation (error out).